pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -1169,6 +1169,25 @@ def test_dti_isub_timedeltalike(self, tz_naive_fixture, two_hours):
         rng -= two_hours
         tm.assert_index_equal(rng, expected)
 
+    def test_dt64arr_add_sub_td64_nat(self, box, tz_naive_fixture):
```

Comment:
can you parameterize other with pd.NaT as well here (I bet we already have a test for that, can you consolidate)

Reply:
Because this is a timedelta64 and pd.NaT is in most cases datetime-like.  The "most cases" part makes it especially hairy.